feature-libs/product-configurator/rulebased/components/attribute/price-change/configurator-attribute-price-change.service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { ObjectComparisonUtils } from '@spartacus/core';
+import { ConfiguratorRouterExtractorService } from '@spartacus/product-configurator/common';
+import { EMPTY, filter, Observable, of, switchMap, tap } from 'rxjs';
+import { ConfiguratorCommonsService } from '../../../core/facade/configurator-commons.service';
+import { Configurator } from '../../../core/model/configurator.model';
+
+@Injectable()
+export class ConfiguratorAttributePriceChangeService {
+  protected configuratorRouterExtractorService = inject(
+    ConfiguratorRouterExtractorService
+  );
+  protected configuratorCommonsService = inject(ConfiguratorCommonsService);
+  protected isInitialRendering: boolean = true;
+  protected lastAttributeSupplement:
+    | Configurator.AttributeSupplement
+    | undefined;
+  protected valuePrices: { [key: string]: Configurator.PriceDetails } = {};
```

Comment:
Let's give the key a more semantic name, for easier understanding what data we use for indexing prices:
```suggestion
  protected valuePrices: { [valueName: string]: Configurator.PriceDetails } = {};
```